src/convert.rs

```diff
@@ -6,24 +6,73 @@ use alloc::{
     string::{String, ToString},
     vec::Vec,
 };
+use core::fmt;
 
 use crate::{cid::Cid, ipld::Ipld};
 
 #[cfg(feature = "std")]
-use crate::error::{Error, TypeError, TypeErrorType};
+//use crate::error::{Error, TypeError, TypeErrorType};
+use crate::error::KindErrorType;
+
+/// Error used for converting from and into [`crate::ipld::Ipld`].
+#[derive(Clone, Debug)]
+pub enum ConversionError {
+    /// Error when the IPLD kind wasn't the one we expected.
+    WrongIpldKind {
+        /// The expected type.
+        expected: KindErrorType,
+        /// The actual type.
+        found: KindErrorType,
+    },
+    /// Error when the given Ipld kind cannot be converted into a certain value type.
+    FromIpld {
+        /// The IPLD kind trying to convert from.
+        from: KindErrorType,
+        /// The type trying to convert into.
+        into: &'static str,
```

Comment:
Can we not use kinds here as well?

Also, I'm not entirely sure how this error and the previous ones differ.

Reply:
This error is created by the `derive_try_from_*` macros. It's for converting from `Ipld` into native Rust types. Here's e.g. the call from `Ipld::Float` to and `f64`:  https://github.com/ipld/rust-ipld-core/blob/6d70de0114daa15e4062b57c466fbc5875ed3c64/src/convert.rs#L204

The `into` field is a string, so that the error can contain which native Rust type we tried to convert into. So it's not an Ipld Kind.

Reply:
Ah. Maybe we can use a TypeID? Although I'm not sure if that works (or is even worth it).

Reply:
Oh, I haven't thought of `TypeId`, that's even nicer (and more expressive than just  as string).